src/gui/virtualstudio.cpp

```diff
@@ -1224,38 +1234,30 @@ void VirtualStudio::refreshStudios(int index, bool signalRefresh)
 void VirtualStudio::refreshDevices()
 {
 #ifdef RT_AUDIO
-    if (!m_vsAudioInterface.isNull()) {
-        m_vsAudioInterface->closeAudio();
-        setAudioReady(false);
-    }
-
-    refreshRtAudioDevices();
-    validateDevicesState();
```

Comment:
Why did `refreshRtAudioDevices()` and `validateDevicesState()` get moved to within `restartAudio`? Just seems a little weird to me that it wouldn't be handled directly by this function.

Reply:
I moved these to `startAudio` because:
* both need to be called before the rest of the code in `startAudio` such as `setInputDevice`
* `restartAudio` calls `startAudio` and `restartAudio` is already called from here;  there is no need to do it twice
* `startAudio` is called via other paths than just `refreshDevices`

Reply:
Note also the removal of `refreshDevices` from the constructor. That was causing it to scan and initialize audio immediately at startup, before it tried to do anything. With this PR, the audio initialization is being deferred until after authentication succeeds, so that we can properly report on what is happening when and have a better shot at tracking down the root cause of issues.